### PR TITLE
[Keyboard] Add ZTBoards keyboards

### PR DESCRIPTION
**Pull request by phlopsuh**
## Description

- Added ZTBoards folder for the After and Noon keyboards

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ZTBoards/after/after.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+#include "after.h"
+
+void matrix_init_kb(void) {
+    // put your keyboard start-up code here
+    // runs once when the firmware starts up
+
+    matrix_init_user();
+    led_init_ports();
+}
```
Comment (zvecr) [inline keyboards/ZTBoards/after/after.c:25]:
```suggestion
```

File: keyboards/ZTBoards/after/after.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#ifndef AFTER_H
+#define AFTER_H
+
+#include "quantum.h"
+//#define XXX KC_NO
+//include the above later to optimize after.h file
+//lets me use xxx instead of kc_no
+
```
Comment (zvecr) [inline keyboards/ZTBoards/after/after.h:8]:
```suggestion
#pragma once

#include "quantum.h"

```

File: keyboards/ZTBoards/after/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#include "after.h"
```
Comment (zvecr) [inline keyboards/ZTBoards/after/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/ZTBoards/after/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+   *    main layer:
+   *       CW: move mouse right
+   *      CCW: move mouse left
+   *    other layers:
+   *       CW: = (equals/plus - increase slider in Adobe products)
+   *      CCW: - (minus/underscore - decrease slider in adobe products)
+   * Thank you to imchipwood/dumbpad for defining this. 
+   */
+  if (index == 0) {
+    switch (biton32(layer_state)) {
```
Comment (zvecr) [inline keyboards/ZTBoards/after/keymaps/default/keymap.c:35]:
```suggestion
    switch (get_highest_layer(layer_state)) {
```

File: keyboards/ZTBoards/after/keymaps/ellicose/keymap.c [hunk 5]
```diff
@@ -0,0 +1,55 @@
+#include "after.h"
```
Comment (zvecr) [inline keyboards/ZTBoards/after/keymaps/ellicose/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/ZTBoards/after/keymaps/ellicose/keymap.c [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+   *    main layer:
+   *       CW: move mouse right
+   *      CCW: move mouse left
+   *    other layers:
+   *       CW: = (equals/plus - increase slider in Adobe products)
+   *      CCW: - (minus/underscore - decrease slider in adobe products)
+   * Thank you to imchipwood/dumbpad for defining this. 
+   */
+  if (index == 0) {
+    switch (biton32(layer_state)) {
```
Comment (zvecr) [inline keyboards/ZTBoards/after/keymaps/ellicose/keymap.c:35]:
```suggestion
    switch (get_highest_layer(layer_state)) {
```

File: keyboards/ZTBoards/after/keymaps/phlop/keymap.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS,          KC_TRNS,                   KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,          KC_TRNS
+		)
+};
+
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    switch(biton32(layer_state)) {
```
Comment (zvecr) [inline keyboards/ZTBoards/after/keymaps/phlop/keymap.c:25]:
```suggestion
    switch(get_highest_layer(layer_state)) {
```

File: keyboards/ZTBoards/after/keymaps/phlop/keymap.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
+#include "after.h"
```
Comment (zvecr) [inline keyboards/ZTBoards/after/keymaps/phlop/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/ZTBoards/after/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+A 65% keyboard made by ZTBoards and run in a Geekhack group buy.
+
+Keyboard Maintainer: ZTBoards 
+Hardware Supported: After QMK PCB 
+Hardware Availability: //update with geekhack link 
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ZTBoards/After:default
```
Comment (zvecr) [inline keyboards/ZTBoards/after/readme.md:11]:
```suggestion
    make ztboards/after:default
```

File: keyboards/ZTBoards/after/rules.mk [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE  = yes  # Mouse keys(+4700)
+EXTRAKEY_ENABLE  = yes  # Audio control and System control(+450)
+CONSOLE_ENABLE   = no   # Console for debug(+400)
+COMMAND_ENABLE   = no   # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE      = yes  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
+AUDIO_ENABLE     = no
+RGBLIGHT_ENABLE  = no
```
Comment (zvecr) [inline keyboards/ZTBoards/after/rules.mk:26]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```

File: keyboards/ZTBoards/noon/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,20 @@
+#include "noon.h"
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/ZTBoards/noon/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,38 @@
+// Default layout for Noon
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/keymaps/via/keymap.c:1]:
```suggestion
```

File: keyboards/ZTBoards/noon/noon.c [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+#include "noon.h"
+
+void matrix_init_kb(void) {
+    // put your keyboard start-up code here
+    // runs once when the firmware starts up
+
+    matrix_init_user();
+    led_init_ports();
+}
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/noon.c:25]:
```suggestion
```

File: keyboards/ZTBoards/noon/noon.h [hunk 14]
```diff
@@ -0,0 +1,23 @@
+#ifndef NOON_H
+#define NOON_H
+
+#include "quantum.h"
+//#define XXX KC_NO
+//include the above later to optimize noon.h file.
+//lets me use xxx instead of kc_no
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/noon.h:7]:
```suggestion
#pragma once

#include "quantum.h"
```

File: keyboards/ZTBoards/noon/noon.h [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+	K400, K401, K402,             K405,                         K410, K411,       K413,       K415  \
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014,  K015 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  KC_NO, K115 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  KC_NO, KC_NO, K215 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  KC_NO, K313,  KC_NO, K315 }, \
+	{ K400,  K401,  K402,  KC_NO, KC_NO, K405,  KC_NO, KC_NO, KC_NO, KC_NO, K410,  K411,  KC_NO, K413,  KC_NO, K415 }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/noon.h:23]:
```suggestion
```

File: keyboards/ZTBoards/noon/readme.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+
+A 65% keyboard made by ZTBoards and run in a Geekhack group buy.
+
+Keyboard Maintainer: ZTBoards  
+Hardware Supported: Noon QMK PCB 
+Hardware Availability: //update with geekhack link 
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ZTBoards/Noon:default
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/readme.md:11]:
```suggestion
    make ztboards/noon:default
```

File: keyboards/ZTBoards/noon/rules.mk [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE  = yes  # Mouse keys(+4700)
+EXTRAKEY_ENABLE  = yes  # Audio control and System control(+450)
+CONSOLE_ENABLE   = no   # Console for debug(+400)
+COMMAND_ENABLE   = no   # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE      = yes  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
+AUDIO_ENABLE     = no
+RGBLIGHT_ENABLE  = no
```
Comment (zvecr) [inline keyboards/ZTBoards/noon/rules.mk:26]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```

File: keyboards/ZTBoards/after/after.h [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+	K400,       K402,             K405,                   K409, K410,       K412, K413,       K415  \
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014,  K015 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  KC_NO, K115 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  KC_NO, KC_NO, K215 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  KC_NO, K312,  K313,  KC_NO, K315 }, \
+	{ K400,  KC_NO, K402,  KC_NO, KC_NO, K405,  KC_NO, KC_NO, KC_NO, K409,  K410,  KC_NO, K412,  K413,  KC_NO, K415 }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/ZTBoards/after/after.h:23]:
```suggestion
```

File: keyboards/ZTBoards/after/config.h [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 16
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS \
+    { B3, F6, F5, D5, B2 }
+#define MATRIX_COL_PINS \
+    { D0, D1, D2, D3, D7, D6, D4, C7, C6, B6, B5, B4, F7, F0, F4, F1 }
```
Comment (fauxpark) [inline keyboards/ZTBoards/after/config.h:37]:
```suggestion
#define MATRIX_ROW_PINS { B3, F6, F5, D5, B2 }
#define MATRIX_COL_PINS { D0, D1, D2, D3, D7, D6, D4, C7, C6, B6, B5, B4, F7, F0, F4, F1 }
```

File: keyboards/ZTBoards/after/config.h [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+#define UNUSED_PINS
+#define ENCODERS_PAD_A { B1 }
+#define ENCODERS_PAD_B { B0 }
+#define ENCODER_RESOLUTION 4 //"Additionally, the resolution, which defines how many pulses the encoder registers between each detent, can be defined"
```
Comment (fauxpark) [inline keyboards/ZTBoards/after/config.h:41]:
```suggestion
```

File: keyboards/ZTBoards/after/readme.md [hunk 21]
```diff
@@ -0,0 +1,15 @@
+# After
+
+A 65% keyboard made by ZTBoards and run in a Geekhack group buy.
+
+Keyboard Maintainer: ZTBoards 
+Hardware Supported: After QMK PCB 
+Hardware Availability: //update with geekhack link
```
Comment (fauxpark) [inline keyboards/ZTBoards/after/readme.md:7]:
```suggestion
* Keyboard Maintainer: ZTBoards
* Hardware Supported: After QMK PCB
* Hardware Availability: //update with geekhack link
```

File: keyboards/ZTBoards/noon/config.h [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 16
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS \
+    { B5, D5, D3, B1, F0 }
+#define MATRIX_COL_PINS \
+    { D2, D1, D0, D4, D6, B2, D7, B4, B6, C6, C7, F7, F6, F5, F4, F1 }
```
Comment (fauxpark) [inline keyboards/ZTBoards/noon/config.h:37]:
```suggestion
#define MATRIX_ROW_PINS { B5, D5, D3, B1, F0 }
#define MATRIX_COL_PINS { D2, D1, D0, D4, D6, B2, D7, B4, B6, C6, C7, F7, F6, F5, F4, F1 }
```

File: keyboards/ZTBoards/noon/readme.md [hunk 23]
```diff
@@ -0,0 +1,15 @@
+# Noon
+
+A 65% keyboard made by ZTBoards and run in a Geekhack group buy.
+
+Keyboard Maintainer: ZTBoards  
+Hardware Supported: Noon QMK PCB 
+Hardware Availability: //update with geekhack link
```
Comment (fauxpark) [inline keyboards/ZTBoards/noon/readme.md:7]:
```suggestion
* Keyboard Maintainer: ZTBoards
* Hardware Supported: Noon QMK PCB
* Hardware Availability: //update with geekhack link
```

File: keyboards/ZTBoards/after/after.h (hunk 2 above)
Comment (phlopsuh) [reply keyboards/ZTBoards/after/after.h:8]: Do I just need to hit commit suggestion for these or do I need to go back and change it on my end?

**Comment by phlopsuh** [comment]
I'm not sure if renaming it a lowercase folder worked, please let me know if there are duplicates. 

For the info.json files, where do I pull the layout section from? KLE?

**Comment by phlopsuh** [comment]
> You seem to have merged QMK's `master` branch into your own, which adds a lot of noise to the pull request and messes with the file history when merged.
> 
> Normally I'd say rebase your master branch, but I tried this locally and got a weird error that I've never seen before and couldn't resolve.
> 
> The quickest way to get this back to a clean state will be to run the following series of commands (though you will lose the incremental history when you do):
> 
> ```shell
> git fetch upstream master  # fetch the state of QMK's master branch
> git reset --hard FETCH_HEAD  # reset this branch to the state of QMK's, which was fetched above
> git checkout c6c2d826c -- keyboards/ztboards/  # checks out the keyboards/ztboards/ directory from the now-current state of this branch
> git commit -m "[Keyboard] Add ZTBoards keyboards"  # commit the staged files
> git push --force-with-lease  # overwrites GitHub's copy of your master branch with your locally-updated master branch
> ```
> 
> This will clean up the pull request so it's only the files needed for the boards you're adding.

$ git push --force-with-lease
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>


This is what I get when I try to run that last line. Would it be easier to close this PR and open a new one? My new branch in my forked repo has the name of my folder in caps and I couldn't change it to lowercase which is why I merged the main branch instead.

**Comment by noroadsleft** [comment]
> $ git push --force-with-lease
> fatal: You are not currently on a branch.
> To push the history leading to the current (detached HEAD)
> state now, use
> 
> ```
> git push origin HEAD:<name-of-remote-branch>
> ```
> 
> This is what I get when I try to run that last line. Would it be easier to close this PR and open a new one? My new branch in my forked repo has the name of my folder in caps and I couldn't change it to lowercase which is why I merged the main branch instead.

Be sure that you're running this from your qmk_firmware directory. The `not currently on a branch` message typically happens when a `git reset` operation deletes the directory that is your current working directory. Notice on the log below, after I run `git reset`, the branch (`pr/9715`) isn't present on the command prompt:

```shell
 noroadsleft  ~/qmk_firmware/keyboards/ztboards (pr/9715)
$ git log -1
commit c6c2d826c6830d61468b4a1e0a7a98bebf5a12d3 (HEAD -> pr/9715)
Merge: 0e37fd056 1cda23b05
Author: Philip Suh [REDACTED]
Date:   Thu Jul 16 19:21:49 2020 -0700

    Merge branch 'master' of https://github.com/phlopsuh/qmk_firmware

 noroadsleft  ~/qmk_firmware/keyboards/ztboards (pr/9715)
$ git reset --hard upstream/master
HEAD is now at 054021e87 [Keyboard] Squiggle (#9387)

 noroadsleft  ~/qmk_firmware/keyboards/ztboards
$
```

**Comment by phlopsuh** [comment]
> > $ git push --force-with-lease
> > fatal: You are not currently on a branch.
> > To push the history leading to the current (detached HEAD)
> > state now, use
> > ```
> > git push origin HEAD:<name-of-remote-branch>
> > ```
> > 
> > 
> > This is what I get when I try to run that last line. Would it be easier to close this PR and open a new one? My new branch in my forked repo has the name of my folder in caps and I couldn't change it to lowercase which is why I merged the main branch instead.
> 
> Be sure that you're running this from your qmk_firmware directory. The `not currently on a branch` message typically happens when a `git reset` operation deletes the directory that is your current working directory. Notice on the log below, after I run `git reset`, the branch (`pr/9715`) isn't present on the command prompt:
> 
> ```shell
>  noroadsleft  ~/qmk_firmware/keyboards/ztboards (pr/9715)
> $ git log -1
> commit c6c2d826c6830d61468b4a1e0a7a98bebf5a12d3 (HEAD -> pr/9715)
> Merge: 0e37fd056 1cda23b05
> Author: Philip Suh [REDACTED]
> Date:   Thu Jul 16 19:21:49 2020 -0700
> 
>     Merge branch 'master' of https://github.com/phlopsuh/qmk_firmware
> 
>  noroadsleft  ~/qmk_firmware/keyboards/ztboards (pr/9715)
> $ git reset --hard upstream/master
> HEAD is now at 054021e87 [Keyboard] Squiggle (#9387)
> 
>  noroadsleft  ~/qmk_firmware/keyboards/ztboards
> $
> ```

Hmm, I'm pretty sure I am in the correct directory as I am running the previous commands in the correct directory. Would you advise against closing this PR and reopening it with the proper branch?

**Comment by noroadsleft** [comment]
> Hmm, I'm pretty sure I am in the correct directory as I am running the previous commands in the correct directory.

I feel like I may have not been clear before: Your current working directory should end in `qmk_firmware` – not `qmk_firmware/keyboards` or `qmk_firmware/keyboards/ztboards` or any other subfolder.

> Would you advise against closing this PR and reopening it with the proper branch?

I wouldn't advise *against* it, but I'm puzzled as to why it works for me but not for you. I ran the exact method I gave you locally twice, and it worked perfectly both times.

**Comment by phlopsuh** [comment]
Yeah, I'm getting the same error when I'm trying the git push command. 

`phili@PhilipSuh MINGW64 /g/keebsoft/qmk_firmware
$ git push --force-with-lease
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>
`

**Pull request closed without merging**